### PR TITLE
Adding tagging to collections.

### PR DESCRIPTION
**Pull request by dannon**
This adds a tag editor to the collections exploded view, and renders 'nametags' for them.  I had it in the primary hdca list item first, like a dataset, but because with the rename is also in the exploded view (at least for now) it makes sense to keep the editables together.  This may also help people actually find that rename, too, until we work on the UX here a bit.  Also fixed a couple bugs in the collections/tag managers to get it working.

### CODE REVIEW

**Comment by nekrut** [comment]
I tested this branch. There are three issues:

- tags do not yet propagate in collections
- history does not refresh (killer for usability)
- the requirement to type `name:tag` is annoying

**Comment by dannon** [comment]
I'm fixing tag propagation for collections now, will be done in a few minutes.
Refresh will be a separate PR (separate item on the improvements card).
And, I'll add another item to the card to look into adding nametags directly, instead of requiring the user to use tags.

**Comment by dannon** [comment]
I think both sorts of collection outputs (explicit and implicit) should inherit tags correctly now.

**Comment by nekrut** [comment]
inheritance works in my test example!

**Comment by dannon** [comment]
@nekrut Great, thanks for testing it again!

**Comment by dannon** [comment]
I went ahead and handled the refresh issues in this branch as well.

**Comment by bgruening** [comment]
:+1: go for it Dannon. It works for me as well.

**Comment by dannon** [comment]
I'll merge dev again and rebuild if it'll make merging this easier for folks :)

**Comment by bgruening** [comment]
please ;)

File: client/galaxy/scripts/mvc/collection/collection-view-edit.js [hunk 1]
```diff
@@ -33,6 +35,7 @@ var CollectionViewEdit = _super.extend(
      */
     initialize : function( attributes ){
         _super.prototype.initialize.call( this, attributes );
+        // this.tagsEditorShown = attributes.tagsEditorShown || false;
```
Comment (guerler) [inline client/galaxy/scripts/mvc/collection/collection-view-edit.js]: Can we just remove this line?

Comment (dannon) [reply client/galaxy/scripts/mvc/collection/collection-view-edit.js]: Sure.

**Comment by guerler** [comment]
Works well for me. 👍 Waiting for tests to complete.

**Pull request merged**